src/features/votes/hooks/useGroupMerkleProof.ts

```diff
@@ -0,0 +1,19 @@
+import { useMemo } from "react";
+import { generateMerkleLeaf } from "../utils/merkleUtils";
+import { useGroupMerkleTree } from "./useGroupMerkleTree";
+
+export const useGroupMerkleProof = (
+  proposalIds: string[],
+  vote: {
+    voter: `0x${string}`;
+    vp: number;
+  },
+) => {
+  const tree = useGroupMerkleTree(proposalIds);
+
+  return useMemo(() => {
+    if (!tree) return;
```

Comment:
```suggestion
    if (!tree) return [];
```

Reply:
As discussed, returning undefined is intentional